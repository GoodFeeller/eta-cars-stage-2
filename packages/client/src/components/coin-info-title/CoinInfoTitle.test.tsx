import { render, screen } from '@testing-library/react'
import CoinInfoTitle from './CoinInfoTitle'
import { BrowserRouter } from 'react-router-dom'
import userEvent from '@testing-library/user-event'

describe('CoinInfoTitle test', () => {
  test('Snapshot test', () => {
    render(
      <BrowserRouter>
        <CoinInfoTitle />
      </BrowserRouter>,
    )
    const title = screen.queryByTestId('title')
    expect(title).toMatchSnapshot()
  })
  test('Link test, url must be localhost/', () => {
    render(
      <BrowserRouter>
        <CoinInfoTitle />
      </BrowserRouter>,
    )
    const link = screen.queryByText('Back') as HTMLElement
    userEvent.click(link)
    expect(document.location.href).toContain('http://localhost/')
  })
})
