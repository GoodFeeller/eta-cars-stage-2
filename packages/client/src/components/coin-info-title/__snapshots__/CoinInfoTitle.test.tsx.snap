// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinInfoTitle test Snapshot test 1`] = `
<div
  class="body"
  data-testid="title"
>
  <a
    class="link"
    href="/"
  >
    Back
  </a>
  <span>
    COIN INFO
  </span>
</div>
`;
